pkg/openstack/loadbalancer.go

```diff
@@ -1137,8 +1126,64 @@ func (lbaas *LbaasV2) ensureOctaviaPool(lbID string, listener *listeners.Listene
 	return pool, nil
 }
 
+func (lbaas *LbaasV2) buildPoolCreateOpt(listenerProtocol string, service *corev1.Service, svcConf *serviceConfig) v2pools.CreateOpts {
+	// By default, use the protocol of the listener
+	poolProto := v2pools.Protocol(listenerProtocol)
+	if svcConf.enableProxyProtocol {
+		poolProto = v2pools.ProtocolPROXY
+	} else if svcConf.keepClientIP && poolProto != v2pools.ProtocolHTTP {
+		klog.V(4).Infof("Forcing to use %q protocol for pool because annotation %q is set", v2pools.ProtocolHTTP, ServiceAnnotationLoadBalancerXForwardedFor)
+		poolProto = v2pools.ProtocolHTTP
+	}
+
+	affinity := service.Spec.SessionAffinity
+	var persistence *v2pools.SessionPersistence
+	switch affinity {
+	case corev1.ServiceAffinityNone:
+		persistence = nil
+	case corev1.ServiceAffinityClientIP:
+		persistence = &v2pools.SessionPersistence{Type: "SOURCE_IP"}
+	}
+
+	lbmethod := v2pools.LBMethod(lbaas.opts.LBMethod)
+	return v2pools.CreateOpts{
+		Protocol:    poolProto,
+		LBMethod:    lbmethod,
+		Persistence: persistence,
+	}
+}
+
+//buildBatchUpdateMemberOpts returns v2pools.BatchUpdateMemberOpts array for Services and Nodes alongside a list of member names
+func (lbaas *LbaasV2) buildBatchUpdateMemberOpts(port corev1.ServicePort, nodes []*corev1.Node, svcConf *serviceConfig) ([]v2pools.BatchUpdateMemberOpts, sets.String, error) {
+	var members []v2pools.BatchUpdateMemberOpts
+	newMembers := sets.NewString()
+
+	for _, node := range nodes {
+		addr, err := nodeAddressForLB(node)
+		if err != nil {
+			if err == ErrNotFound {
```

Comment:
ErrNotFound is not returned by nodeAddressForLB , it should be ErrNoAddressFound?

Reply:
The error `ErrNotFound` is taken over from https://github.com/kubernetes/cloud-provider-openstack/pull/1365/files/8ebc677f47624632ddea68def6f230c70b3a2a46#diff-89af30f72ae5c8aefb464330d654b95dc41544d558d9c04c9e728a8ea9a94793L1051, so it doesn't change the expected return signature of the higher-level function call. Changing the Error type would change it. 

I would like to maintain the behaviour and optionally change in an other PR specifically.